csrc/multidevice/utils.cpp

```diff
@@ -220,9 +183,92 @@ void shardAllLike(TensorView* ref, std::vector<TensorView*> tvs) {
     tv->setDeviceMesh(ref->getDeviceMesh());
   }
   if (!tvs.empty()) {
-    scheduler_utils::parallelizeAllLike(ref, tvs, {ParallelType::DIDx});
+    scheduler_utils::parallelizeAllLike(
+        ref, tvs, {ParallelType::DIDx, ParallelType::Serial});
+  }
+}
+
+// TODO: We can either reshard the inputs of a resharding expression or
+// the outputs. Currently, we reshard the outputs when there is only one
+// input, otherwise we reshard the inputs. This heuristic should be smarter
+// and attempt to minimize communication.
+bool reshardAfter(Expr* expr) {
```

Comment:
```suggestion
bool shouldReshardAfter(Expr* expr) {
```

Naming nit: `reshardAfter` gives readers the wrong impression that the function actually inserts resharding.